src/ITCC.YandexSpeeckKitClient/Utils/EnumValueNameHelper.cs

```diff
@@ -0,0 +1,147 @@
+﻿// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using ITCC.YandexSpeeckKitClient.Attributes;
+
+namespace ITCC.YandexSpeeckKitClient.Utils
+{
+    internal static class EnumValueNameHelper
+    {
+        #region public
+
+        public static string GetEnumString(this object element)
+        {
+            var attributes = GetAttributes(element);
+            return attributes == null ? null : string.Join(Separator, attributes.Select(attr => attr.Name));
+        }
+        public static object GetEnumElementByName(this string name, Type enumType)
+        {
+            if (name == null)
+                return null;
+
+            var dictionary = GetInfoDictionaty(enumType);
+            if (dictionary == null)
+                return null;
+
+            if (IsFlagEnum(enumType))
+            {
+                var list = new List<object>();
+                var stringValues = name.Split(new[] { Separator }, StringSplitOptions.RemoveEmptyEntries);
+                foreach (var stringValue in stringValues)
+                {
+                    if (dictionary.Values.Any(value => value?.Name == stringValue))
+                        list.Add(dictionary.First(keyValuePair => keyValuePair.Value.Name == stringValue).Key);
+                    else
+                        return null;
+                }
+
+                return list.Aggregate(0, (current, elem) => current | (int)elem);
+            }
+
+            return dictionary.Values.All(value => value?.Name != name) ? null : dictionary.First(keyValuePair => keyValuePair.Value.Name == name).Key;
```

Comment:
operator `?` cannot be applied to operand of type `KeyValuePair<TKey,TValue>` (struct)